src/Sync.js

```diff
@@ -400,9 +404,31 @@ function determineExtension(path) {
     const parts = path.split(".");
     return parts[parts.length - 1];
 }
-// extractID uses regex to exract the id from a string
-function extractID(line) {
-  return line.match(/(?<=\bSNIPSTART\s)(\w+-\w+)+(\w+)/g)[0];
+
+// See: https://stackoverflow.com/questions/3446170/escape-string-for-use-in-javascript-regex
```

Comment:
So we are escaping the string first, before attempting a match - because of the added `{ }` in the SNIP placeholders?

Reply:
No, we're escaping it because we want a regex safe version of the different markers which I predict we'll want to make configurable at some point, e.g `writeStart` could be `[SNIPSTART` which if not escaped will result in an invalid regexp (and will throw `SyntaxError`).